admin-dev/themes/new-theme/js/app/components/checkbox.vue

```diff
@@ -0,0 +1,87 @@
+<!--**
```

Comment:
@PierreRambaud  nice! 

However since it is in `app/components/checkbox.vue` then I believe this component itself should be highly reusable. I'm not tested it but I know that this method should work:
```
<input
type="checkbox"
v-bind="$attrs"
v-on=“$listeners"
/>
</WithErrorMessage>
</template>
<script>
export default {
inheritAttrs: false,
// ...
}
</script>
```
this part:
```
v-bind="$attrs"
v-on=“$listeners"
```
Both props and attributes, as well
as all listeners will be passed to
this element instead.

and **inheritAttrs: false** Prevent Vue from assigning
attributes to top-level element

after having this code added in your page specific javascript you can create permissions-checkbox.vue file as a new component which will use this highly reusable component . This will help us to decouple the checkbox component and use it in every case.Wdyt?

Reply:
I think we must improve component depending on what we need. 
For now, we don't need to add must complexity, things we don't need, and we can't tests (because we don't have tests and we don't use these kinds of behaviors ^^).
We mustn't wasting time on things we can improve later :wink:

Reply:
well yes I can agree with you but imagine I will need to add another attr to my input element, or maybe to fire some more events like focus etc... I will need to add everything using properties. This is just a two samples but there might be more. 

But again I'm using words **it might be** . However if we face some kind of refactoring of this global component don't we introduce bc breaks later on? Don't you think we need to make it as close as native checkbox element can get us without passing a lot of properties and adding many conditions to it? :thinking:

Reply:
You can improve this component without adding breaking changes.
And because it's html / vueJs, you're also able to add your own component / use simple checkbox. 
If you want to make it full customisable , you will add and work for something that's currently not needed and maybe will never used.

Reply:
Okey , sounds good to me :)